modules/dhcp/providers/server/isc.rb

```diff
@@ -146,31 +146,34 @@ def loadSubnetData subnet
       initialize_memory_store_with_dhcp_records(parse_config_and_leases_for_records)
     end
 
+    SUBNET_BLOCK_REGEX = /subnet\s+([\d\.]+)\s+netmask\s+([\d\.]+)\s*\{([\w-]+\{[^{}]*\}|[\w-][^{}]*;)*\}/
```

Comment:
This removes the \s that was permitted between the two curly braces in the earlier version, so for example, this now fails:

    assert "subnet 192.168.1.0 netmask 255.255.255.128 {pool\n{range abc;}}".match(Proxy::DHCP::ISC::SUBNET_BLOCK_REGEX)

It needs to be rather liberal in its accepting of whitespace.  Perhaps we should just match the start and end of the block with curly braces?  What's in the middle doesn't matter too much as it's parsed below.